File: app/src/main/java/com/hiddenramblings/tagmo/amiibo/games/GameTitles.java
package com.hiddenramblings.tagmo.amiibo.games;

import com.hiddenramblings.tagmo.amiibo.Amiibo;

import org.json.JSONArray;
import org.json.JSONException;

import java.util.ArrayList;

public class GameTitles {

    public final GamesManager manager;
    public final String name;
    private final ArrayList<Long> gameIds = new ArrayList<>();

    public GameTitles(GamesManager manager, String name, JSONArray jsonArray) {
        this.manager = manager;
        this.name = name;
        for (int i = 0; i < jsonArray.length(); i++) {
            try {
                String gameId = jsonArray.getString(i);
                gameIds.add(GamesManager.hexToId("0x" + gameId));
            } catch (JSONException e) {
                e.printStackTrace();
            }
        }
    }
}

File: app/src/main/java/com/hiddenramblings/tagmo/amiibo/games/GameTitles.kt
package com.hiddenramblings.tagmo.amiibo.games

import com.hiddenramblings.tagmo.eightbit.io.Debug
import org.json.JSONArray
import org.json.JSONException

class GameTitles(val manager: GamesManager, val name: String, jsonArray: JSONArray) {
    private val gameIds = ArrayList<Long>()

    init {
        for (i in 0 until jsonArray.length()) {
            try {
                val gameId = jsonArray.getString(i)
                gameIds.add(java.lang.Long.decode("0x$gameId"))
            } catch (e: JSONException) {
                Debug.warn(e)
            }
        }
    }
}